zero_functional.nim

```diff
@@ -284,6 +296,26 @@ proc inlineFilter(ext: ExtNimNode, toSeq: bool = false): ExtNimNode {.compileTim
           nil
   result = ext
 
+proc inlineSub(ext: ExtNimNode, toSeq: bool = false): ExtNimNode {.compileTime.} =
```

Comment:
I hope sub is a good name
sub just refers to filter using the internal idx variable

Reply:
`sub` seems good